dagger-functions/src/main/java/io/odpf/dagger/functions/udfs/scalar/dart/types/Cache.java

```diff
@@ -31,8 +31,7 @@ public boolean hasExpired(int refreshIntervalInHours) {
         if (timeOfCaching == null) {
             return true;
         }
-        Date currentTime = new Date();
-        Date timeOfExpire = DateUtils.addHours(currentTime, refreshIntervalInHours);
+        Date timeOfExpire = DateUtils.addHours(timeOfCaching, refreshIntervalInHours);
         return timeOfCaching.after(timeOfExpire);
```

Comment:
shouldn't this be `time.now().after(timeOfExpire)`, basically the current statement will also never return true

can you also add corresponding tests? I think this was missed primarily because of that

and you will need to bump up version in version.txt file post this

Reply:
yes, you are right it should be compared with the current time.